.github/workflows/static_analysis.yml

```diff
@@ -28,3 +28,15 @@ jobs:
       - uses: scottbrenner/cfn-lint-action@v2
         with:
           command: cfn-lint --info --ignore-checks W3002 --template **/cloudformation.yaml
+
+  statelint:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: ruby/setup-ruby@v1
+        with:
+          ruby-version: 2.7
+      - run: gem install statelint
+      - run: |
+          sed -i 's/"Resource": "${.*}"/"Resource": "foo:bar"/' step-function/step-function.json
```

Comment:
What's the purpose of this `sed` command?

Reply:
We use [Definition Substitutions](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-stepfunctions-statemachine.html#cfn-stepfunctions-statemachine-definitionsubstitutions) to inject the lambda ARNs to call into the step function template. Those lines look like this in the template:

```
"Resource": "${VerifyLambdaArn}",
```

Unfortunately, statelint isn't substitution-aware and complains:
```
$ statelint step-function.json
5 errors:
 State Machine.States.Verify.Resource is "${VerifyLambdaArn}" but should be A URI
 State Machine.States.Ingest.Resource is "${IngestLambdaArn}" but should be A URI
 State Machine.States.Echo10Construction.Resource is "${Echo10ConstructionLambdaArn}" but should be A URI
 State Machine.States.Echo10ToCMR.Resource is "${CmrActivity}" but should be A URI
 State Machine.States.Notify.Resource is "${NotifyLambdaArn}" but should be A URI
```

The `sed` command changes those `Resource` values to valid URIs before linting:
```
"Resource": "foo:bar",
```

Reply:
Copied from HyP3 at https://github.com/ASFHyP3/hyp3/blob/develop/.github/workflows/static-analysis.yml#L67

Reply:
Makes sense!